client/src/hooks/graphql/use-with-recommended-action.tsx

```diff
@@ -293,13 +306,35 @@ function parseMentorConditions(
   });
   const incompleteRequirement = categories
     .filter((c) => c.isRequired)
-    .find((c) => c.answers.find((a) => a.status === Status.INCOMPLETE));
+    .find((c) =>
+      c.answers.find(
+        (a) =>
+          !isAnswerComplete(
+            a,
+            getValueIfKeyExists(a.question, mentorQuestions)?.question?.name,
+            mentor.mentorType
+          )
+      )
+    );
   const firstIncomplete = categories.find((c) =>
-    c.answers.find((a) => a.status === Status.INCOMPLETE)
+    c.answers.find(
+      (a) =>
+        !isAnswerComplete(
+          a,
+          getValueIfKeyExists(a.question, mentorQuestions)?.question?.name,
+          mentor.mentorType
+        )
+    )
   );
 
   const completedAnswers =
-    mentor?.answers.filter((a) => a.status === Status.COMPLETE).length || 0;
+    mentor?.answers.filter((a) =>
```

Comment:
I'm pretty sure this code isn't reached until the mentor exists, so no need for ?